ledger/participant-integration-api/src/main/scala/platform/store/backend/common/InitHookDataSourceProxy.scala

```diff
@@ -38,6 +38,11 @@ private[backend] case class InitHookDataSourceProxy(
     } catch {
       case t: Throwable =>
         logger.warn(s"Init hook execution failed", t)
+        try {
```

Comment:
You could also use `Using`:
```
Using(connectionBody) { connection => 
  initHook(connection)
}.fold(t => throw t, identity) 
```

- `Connection` is AutoCloseable, so should be closed by `Using`
- `Using` also handles exceptions in `connectionBody` (unlike the current code)

Not sure what happens if just the closing of the connection fails though, we probably do not care about that error.

Reply:
It's a lot more easier to read in the form @rautenrieth-da proposed.

Reply:
won't Using actually close the connection anyway?